api/src/controllers/all-docs.js

```diff
@@ -0,0 +1,5 @@
+const allDocs = require('../services/all-docs');
+
+module.exports = {
+  request: (req, res) => allDocs.filterOfflineRequest(req, res)
+};
```

Comment:
This is kind of breaking the pattern I was going for with routes, controllers, and services. The idea for the controller was to pull everything out of the request so you could pass just the bits you need to the services, and then write whatever is returned from the service back to the response. This way the service is easy to test and reuse because all the endpoint specific stuff is handed by the controller.

I don't feel like this controller provides any value currently because it's just passing everything through to the service. There are a few options for fixing this...

1) Inline the service in the controller - this is how the code used to be before the controller/service pattern was introduced.
2) Change the service to have a function which takes the userCtx and requested IDs and returns the IDs the user is allowed to see. Then have the controller (or another service) fetch those IDs.

This also applies to the other controller changes in this PR.

I'm happy to talk more about what I see as the reason for doing controllers this way, or how I can better communicate the pattern. Let me know!

Reply:
I've refactored the controllers/services to follow the pattern. Do you think it's better now ?

Reply:
Yes! That looks great. What do you think?